File: src/main/java/eterea/core/api/rest/model/Negocio.java
/**
 * 
 */
package eterea.core.api.rest.model;

import java.io.Serializable;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

/**
 * @author daniel
 *
 */
@Data
@Table
@Entity
@EqualsAndHashCode(callSuper = false)
@NoArgsConstructor
@AllArgsConstructor
public class Negocio extends Auditable implements Serializable {
	/**
	 * 
	 */
	private static final long serialVersionUID = 816078638050323268L;

	@Id
	@Column(name = "neg_id")
	private Integer negocioId;

	@Column(name = "neg_nombre")
	private String nombre;

	@Column(name = "neg_id_real")
	private Integer negocioIdReal;

	@Column(name = "neg_ip")
	private String ipAddress;

	@Column(name = "neg_db")
	private String database;

	@Column(name = "neg_user")
	private String user;

	@Column(name = "neg_transfstock")
	private Byte transferenciaStock = 0;

	@Column(name = "neg_transfvalor")
	private Byte transferenciaValor = 0;

	private String backendServer;
	private String backendPort;

}

File: src/main/java/eterea/core/api/rest/kotlin/model/Negocio.kt
package eterea.core.api.rest.kotlin.model

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id

@Entity
data class Negocio(

    @Id
    @Column(name = "neg_id")
    var negocioId: Int? = null,

    @Column(name = "neg_nombre")
    var nombre: String = "",

    @Column(name = "neg_id_real")
    var negocioIdReal: Int = 0,

    @Column(name = "neg_ip")
    var databaseIpVpn: String = "",

    var databaseIpLan: String = "",

    @Column(name = "neg_db")
    var database: String? = null,

    @Column(name = "neg_user")
    var user: String? = null,

    @Column(name = "neg_transfstock")
    var transferenciaStock: Byte = 0,

    @Column(name = "neg_transfvalor")
    var transferenciaValor: Byte = 0,

    @Column(name = "backend_server")
    var backendIpVpn: String = "",

    var backendIpLan: String = "",
    var backendPort: String = "",
    var facturaServer: String = "",
    var facturaPort: String = "",
    var hasGateway: Byte = 0

) : Auditable()
